dspace-api/src/main/java/org/dspace/identifier/DOI.java

```diff
@@ -67,11 +70,20 @@ public void setDoi(String doi) {
     public DSpaceObject getDSpaceObject() {
         return dSpaceObject;
     }
-
+    
     public void setDSpaceObject(DSpaceObject dSpaceObject) {
         this.dSpaceObject = dSpaceObject;
     }
-
+    
+    public Integer getResourceTypeId() {
+        return this.resourceTypeId;
+    }
+    
+    public void setResourceTypeId(Integer resourceTypeId)
+    {
+        this.resourceTypeId = resourceTypeId;
```

Comment:
I would leave out this setter & do it like ResourcePolicy does it: https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/authorize/ResourcePolicy.java#L170

Because having a separate setter for DSpaceObject & one for type will lead to issues.